packages/node_modules/@ciscospark/react-container-notifications/src/selectors.js

```diff
@@ -1,20 +1,34 @@
 import {createSelector} from 'reselect';
 
+import {NOTIFICATION_TYPE_CALL, NOTIFICATION_TYPE_POST} from './actions';
+
 const getActivities = (state) => state.conversation.activities; // eslint-disable-line func-style
 const getNotifications = (state) => state.notifications.items; // eslint-disable-line func-style
 const getAvatars = (state) => state.avatar.get(`items`).toJS(); // eslint-disable-line func-style
+const getIncomingCall = (state) => state.media.getIn([`incomingCall`, `call`]);
 
 const getUnsentNotifications = createSelector(
-  [getActivities, getNotifications, getAvatars],
-  (activities, notifications, avatars) =>
+  [getActivities, getNotifications, getAvatars, getIncomingCall],
+  (activities, notifications, avatars, incomingCall) =>
     notifications
       .map((notification) => {
-        const notificationActivity = findActivityWithUrl(activities, notification.notificationId);
-        return Object.assign({}, notification, {
-          username: notificationActivity.actor.displayName,
-          message: notificationActivity.object.displayName,
-          avatar: avatars[notificationActivity.actor.id]
-        });
+        if (notification.type === NOTIFICATION_TYPE_POST) {
+          const notificationActivity = findActivityWithUrl(activities, notification.notificationId);
+          return Object.assign({}, notification, {
+            username: notificationActivity.actor.displayName,
+            message: notificationActivity.object.displayName,
+            avatar: avatars[notificationActivity.actor.id]
+          });
+        }
+        else if (notification.type === NOTIFICATION_TYPE_CALL) {
+
+          return Object.assign({}, notification, {
+            username: incomingCall.from.person.name,
+            message: `Incoming Call`,
```

Comment:
Yea, I think a good way to refactor it would be to have the components generating the notifications be in charge of creating the notification strings.